src/descriptors/doc.md

```diff
@@ -0,0 +1,82 @@
+# Marina output descriptors template
+The `descriptors` package extends [the original syntax of output script descriptors](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md).
+
+Giving a string template, the `evaluate` function is able to compute:
+* redeem script 
+* witness scripts if needed (the unsigned part of the witness).
+
+## Example
+
+Assuming `xpub0123456789` is one of the Marina accounts master extended public key, and given the following template: 
+
+```
+elp2wsh(asm(xpub0123456789 OP_CHECKSIG))
+```
+
+Marina will creates a `Context` object mapping each `xpub...` in the template to their actual derivation path:
+
+* in case of signing an input: the path equals the one of the utxo.
+* in case of create a new address: the path equals the one of the next address.
+
+Then calling `evaluate` we are able to get our witness script and redeem script:
+
+```ts
+const result = evaluate(marinaContext, "elp2wsh(asm(xpub0123456789 OP_CHECKSIG))")
+const witness = result.witnesses[0] // segwit v0 has 1 witness
+const redeemScript = result.redeemScript // OP_O .... 
+```
+
+> The `result` object value depends on the Context! 
+
+## API
+
+All descriptors can accept `xpub...` token, the public keys are replaced before parsing the template.
+
+### Raw
+
+Expects an hex string as input.
+
+```
+raw(010203)
+```
+
+Returns the hex string in raw as `redeemScript` when evaluated, do not create any witness.
+
+### Asm
+
+Expects Bitcoin ASM script as input.
+
+```
+asm(OP_DUP OP_HASH160 0123456 OP_EQUALVERIFY OP_CHECKSIG)
+```
+
+Returns the hex value of the script as `redeemScript`. Do not create any witness.
+
+### Elp2wsh
+
+It lets to create a segwit v0 script. It expects another descriptor returning a `redeemScript` as input. Often used with `raw` or `asm`:
+
+```
+elp2wsh(raw(0123456789)) 
+```
+
+Returns:
+
+* `raw(0123456789)` redeem script as `witnesses`.
+* the segwit v0 script `(OP_0 + hash(raw(0123456789)))` as `redeemScript`.
+
+### Eltr
+
+It lets to create [P2TR](https://en.bitcoin.it/wiki/Taproot) payments. The descripor expects 2 inputs:
+
+1. The internal taproot public key. This can be replaced by an `xpub` token in order to increment it according to actual derivation path. Will fail if this is not a 64bits hex string.
```

Comment:
we should maybe add `H_POINT` keyword in preprocessing, so we map to unspendable internal pub key for script-path-only P2TR for the three cases